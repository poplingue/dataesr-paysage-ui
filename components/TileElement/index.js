import { Icon, Tile, TileBody } from '@dataesr/react-dsfr';
import PropTypes from 'prop-types';
import dsfrGrid from '../../helpers/imports';

export default function TileElement({ color, title, subTitle, body, onClick }) {
    const { Col, Row, Container } = dsfrGrid();

    return (
        <Tile color={color} className="w-100" horizontal onClick={onClick}>
            <TileBody title={title} description={subTitle}>
                <Container fluid className="w-100">
                    <Row alignItems="top">
                        <Col n="11">
                            <div>{body}</div>
                        </Col>
                        {onClick && (
                            <Col n="1" className="txt-right">
                                <Icon
                                    name="ri-focus-2-fill"
                                    size="2x"
                                    color={color}
                                />
                            </Col>
                        )}
                    </Row>
                </Container>
            </TileBody>
        </Tile>
    );
}

TileElement.defaultProps = {
    color: '#000091',
    checked: false,
};
TileElement.propTypes = {
    color: PropTypes.string,
    title: PropTypes.string.isRequired,
    body: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
    onClick: PropTypes.func.isRequired,
    checked: PropTypes.bool,
    defaultIcon: PropTypes.string,
};
