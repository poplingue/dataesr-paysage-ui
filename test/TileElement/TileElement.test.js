import preloadAll from 'jest-next-dynamic';
import React from 'react';

import TileElement from '../../components/TileElement';
import { fireEvent, render, screen } from '../test-utils';

describe('TileElement component', () => {
    let wrapper;

    beforeAll(async () => {
        await preloadAll();
    });

    beforeEach(() => {
        const func = () => {};

        wrapper = (context = {}, props) =>
            render(
                <TileElement
                    title={props.title}
                    body={props.body || 'body'}
                    subTitle={props.sub || 'subTitle'}
                    onClick={props.onClick || func}
                />,
                context
            );
    });

    it('renders correctly TileElement', () => {
        const props = { title: 'Title 0', checked: false };
        const tree = wrapper({ stateList: { exportMode: false } }, props);
        expect(tree).toMatchSnapshot();
    });

    it('should display Texts', () => {
        wrapper(
            { stateList: { exportMode: true } },
            {
                title: 'Title 1',
                body: 'Bobody',
                subTitle: 'subTitle',
            }
        );
        const title = screen.getByText('Title 1');
        const body = screen.getByText('Bobody');
        expect(title).toBeVisible();
        expect(body).toBeVisible();
    });

    it('should trigger onClick', () => {
        const mockClick = jest.fn();
        wrapper({}, { onClick: mockClick, title: 'Title' });
        const tile = screen.getByText('Title');
        fireEvent.click(tile, {});
        expect(mockClick).toHaveBeenCalledTimes(1);
    });
});
