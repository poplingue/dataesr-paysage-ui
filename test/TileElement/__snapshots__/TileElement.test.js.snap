// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TileElement component renders correctly TileElement 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="fr-tile fr-tile--grey fr-enlarge-link fr-tile--horizontal w-100"
        role="button"
        style="box-shadow: inset 0 0 0 1px var(--border-default-grey),inset 0 -.25rem 0 0 #000091;"
        tabindex="0"
      >
        <div
          class="fr-tile__body"
        >
          <h3
            class="fr-tile__title"
          >
            <a
              aria-disabled="true"
              class="fr-tile__link"
              role="link"
              target="_self"
            >
              Title 0
            </a>
          </h3>
          <p
            class="fr-tile__desc"
          >
            subTitle
          </p>
          <div
            class="fr-container-fluid w-100"
          >
            <div
              class="fr-grid-row fr-grid-row--top"
            >
              <div
                class="fr-col-11"
              >
                <div>
                  body
                </div>
              </div>
              <div
                class="fr-col-1 txt-right"
              >
                <span>
                  <span
                    aria-hidden="false"
                    class="ri-2x icon-left ds-fr--v-middle ri-focus-2-fill"
                    style="color: rgb(0, 0, 145);"
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="fr-tile fr-tile--grey fr-enlarge-link fr-tile--horizontal w-100"
      role="button"
      style="box-shadow: inset 0 0 0 1px var(--border-default-grey),inset 0 -.25rem 0 0 #000091;"
      tabindex="0"
    >
      <div
        class="fr-tile__body"
      >
        <h3
          class="fr-tile__title"
        >
          <a
            aria-disabled="true"
            class="fr-tile__link"
            role="link"
            target="_self"
          >
            Title 0
          </a>
        </h3>
        <p
          class="fr-tile__desc"
        >
          subTitle
        </p>
        <div
          class="fr-container-fluid w-100"
        >
          <div
            class="fr-grid-row fr-grid-row--top"
          >
            <div
              class="fr-col-11"
            >
              <div>
                body
              </div>
            </div>
            <div
              class="fr-col-1 txt-right"
            >
              <span>
                <span
                  aria-hidden="false"
                  class="ri-2x icon-left ds-fr--v-middle ri-focus-2-fill"
                  style="color: rgb(0, 0, 145);"
                />
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
